src/main.rs

```diff
@@ -20,24 +18,40 @@ pub enum Name {
     Quit,
 }
 
-fn main() {
-    let surface = GL33Surface::build_with(|video| video.window("game", 800, 600))
-        .expect("Failed to create surface");
-
-    main_loop(surface);
-}
+fn bind_inputs(sdl: &Sdl) -> input::InputManager<Name> {
+    let mut input = input::InputManager::new(sdl);
 
-fn main_loop(mut surface: GL33Surface) {
-    use lingon::renderer::*;
-    let mut sampler = luminance::texture::Sampler::default();
-    sampler.mag_filter = luminance::texture::MagFilter::Nearest;
-    let mut renderer = Renderer::new(&mut surface, sampler);
-
-    let mut assets = AssetSystem::new();
-    let image = assets.load_image(Path::new("res/coin-gold.png").to_path_buf());
+    input.bind(input::Device::Key(input::Keycode::A), Name::Left);
+    input.bind(input::Device::Key(input::Keycode::D), Name::Right);
+    input.bind(input::Device::Key(input::Keycode::W), Name::Up);
+    input.bind(input::Device::Key(input::Keycode::S), Name::Down);
+    input.bind(input::Device::Key(input::Keycode::Escape), Name::Quit);
+    input.bind(input::Device::Key(input::Keycode::F), Name::PlaySound);
+    input.bind(input::Device::Quit, Name::Quit);
+    input.bind(input::Device::Axis(0, input::Axis::LeftX), Name::Right);
+    input.bind(input::Device::Axis(0, input::Axis::RightY), Name::Up);
 
-    let sheet = renderer.add_sprite_sheet(assets[image].clone(), (16, 16));
+    input
+}
 
+fn main() {
+    // Create the initial game state and input manager.
+    let mut game = lingon::Game::new("game", 800, 600);
+    let mut input = bind_inputs(game.sdl());
+    //XXX The input manager is completely separate since I don't want to store it on the struct,
+    //but it still feels weird.
+
+    // Load an image and a sound.
+    let coin = game.assets.load_image(Path::new("res/coin-gold.png").to_path_buf());
+    let bloop = game.assets.load_audio(Path::new("res/bloop.wav").to_path_buf());
+    //XXX game.load_image("res/coin-gold.png") and game.load_audio("res/bloop.wav")
```

Comment:
See the myriad of other comments on this. We're not solving anything, we will have to do something like this in the sylt bindings anways.